### PR TITLE
CREATE DATABASE & ADD RESOURCE distsql finish

### PR DESCRIPTION
**Pull request by isHuangXin**
ADD RESOURCE dataSource #17939 

In this PR, I finished parsing the CREATE DATABASE and ADD RESOURCE in the yaml file and generating the corresponding distsql statement.
Several situations have been considered in my implementation, which will be explained with screenshots next.

- If url exists, read url directly instead of host, port, db
- If url does not exist, read host, port, db directly
- If properties exist, read connectionTimeoutMilliseconds under properties, etc...
- If properties does not exist, directly read connectionTimeoutMilliseconds, etc...

**Finally, the result of running my code is**

- The yaml file that needs to be converted
image

- The converted DistSQL statement
image

**Help Require**
Before I submit this PR, I try to run the build process start command: `mvn -T 1C clean install`, but I get this bug.
image
image
image

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                                               + System.lineSeparator()
+                                               + ADD_RESOURCE));
+        }
+        stringBuilder.append(ADD_RESOURCE);
+        for (Map.Entry<String, YamlProxyDataSourceConfiguration> entry : databaseResource.entrySet()) {
+            String resourceName = entry.getKey();
+            String url = entry.getValue().getUrl();
+            String username = entry.getValue().getUsername();
+            String password = entry.getValue().getPassword();
+            String connectionTimeoutMilliseconds = null == entry.getValue().getConnectionTimeoutMilliseconds() ? String.format("")
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java:105]: Empty string does not need to use `String.format`

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            String idleTimeoutMilliseconds = null == entry.getValue().getIdleTimeoutMilliseconds() ? String.format("")
+                    : String.format("\"idleTimeoutMilliseconds\"=%s,", entry.getValue().getIdleTimeoutMilliseconds().toString());
+            String maxLifetimeMilliseconds = null == entry.getValue().getMaxLifetimeMilliseconds() ? String.format("")
+                    : String.format("\"getMaxLifetimeMilliseconds\"=%s,", entry.getValue().getMaxLifetimeMilliseconds().toString());
+            String maxPoolSize = null == entry.getValue().getMaxPoolSize() ? String.format("") : String.format("\"maxPoolSize\"=%s,", entry.getValue().getMaxPoolSize().toString());
+            String minPoolSize = null == entry.getValue().getMinPoolSize() ? String.format("") : String.format("\"minPoolSize\"=%s", entry.getValue().getMinPoolSize().toString());
+            stringBuilder.append(String.format(RESOURCES, resourceName, url, username, password, connectionTimeoutMilliseconds,
+                    idleTimeoutMilliseconds, maxLifetimeMilliseconds, maxPoolSize, minPoolSize));
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java:114]: customPoolProps missed.   (in `YamlProxyDataSourceConfiguration`)

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            String resourceName = entry.getKey();
+            String url = entry.getValue().getUrl();
+            String username = entry.getValue().getUsername();
+            String password = entry.getValue().getPassword();
+            String connectionTimeoutMilliseconds = null == entry.getValue().getConnectionTimeoutMilliseconds() ? String.format("")
+                    : String.format("\"connectionTimeoutMilliseconds\"=%s,", entry.getValue().getConnectionTimeoutMilliseconds().toString());
+            String idleTimeoutMilliseconds = null == entry.getValue().getIdleTimeoutMilliseconds() ? String.format("")
+                    : String.format("\"idleTimeoutMilliseconds\"=%s,", entry.getValue().getIdleTimeoutMilliseconds().toString());
+            String maxLifetimeMilliseconds = null == entry.getValue().getMaxLifetimeMilliseconds() ? String.format("")
+                    : String.format("\"getMaxLifetimeMilliseconds\"=%s,", entry.getValue().getMaxLifetimeMilliseconds().toString());
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java:110]: Incorrect key: `getMaxLifetimeMilliseconds`

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                                            + System.lineSeparator()
+                                            + "\tPASSWORD=%s,"
+                                            + System.lineSeparator()
+                                            + "\tPROPERTIES(%s%s%s%s%s)"
+                                            + System.lineSeparator()
+                                            + "),";
+    
+    @Override
+    protected Collection<String> getColumnNames() {
+        return Collections.singleton("converted DistSQL");
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java:58]: Column names are all lowercase and separated by underscores.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        if (null == stringBuilder) {
+            stringBuilder.append(ADD_RESOURCE);
+        } else {
+            stringBuilder.append(String.format(System.lineSeparator()
+                                               + System.lineSeparator()
+                                               + ADD_RESOURCE));
+        }
+        stringBuilder.append(ADD_RESOURCE);
+        for (Map.Entry<String, YamlProxyDataSourceConfiguration> entry : databaseResource.entrySet()) {
+            String resourceName = entry.getKey();
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java:101]: There are too many `entry.getValue()` in the following code, get a dataSourceConfiguration object to use.

**Comment by RaigorJiang** [comment]
Hi @isHuangXin 
Great feature, please also add test case to assert that the output is as expected.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java (hunk 2 above)
Comment (RaigorJiang) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java:114]: Here is a sample for `customPoolProps `:
https://github.com/apache/shardingsphere/issues/17646

**Comment by RaigorJiang** [comment]
@isHuangXin 
Thanks for your work, please complete the following tasks in future PRs:
- Add test cases
- Optimized code, for example, true | false in properties does not require special judgment and is passed as a string
- Fix comments

**Review by RaigorJiang — APPROVED** [review]

**Comment by isHuangXin** [comment]
> @isHuangXin Thanks for your work, please complete the following tasks in future PRs:
> 
> * Add test cases
> * Optimized code, for example, true | false in properties does not require special judgment and is passed as a string
> * Fix comments

Thank you for your patience and guidance, the next PR will fix these issues.

**Pull request merged**